exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -8,210 +7,372 @@ import org.jetbrains.exposed.sql.vendors.FunctionProvider
 import org.jetbrains.exposed.sql.vendors.currentDialect
 import java.math.BigDecimal
 
-fun ExpressionWithColumnType<*>.count() : Function<Int> = Count(this)
+// String Functions
 
-fun Column<*>.countDistinct() : Function<Int> = Count(this, true)
+/** Converts this string expression to lower case. */
+fun <T : String?> Expression<T>.lowerCase(): LowerCase<T> = LowerCase(this)
 
-fun<T:Comparable<T>, S:T?> ExpressionWithColumnType<in S>.min()  : ExpressionWithColumnType<T?> = Min<T, S>(this, this.columnType)
+/** Converts this string expression to upper case. */
+fun <T : String?> Expression<T>.upperCase(): UpperCase<T> = UpperCase(this)
 
-fun<T:Comparable<T>, S:T?> ExpressionWithColumnType<in S>.max() : ExpressionWithColumnType<T?> = Max<T, S>(this, this.columnType)
+fun <T : String?> Expression<T>.groupConcat(
+    separator: String? = null,
+    distinct: Boolean = false,
+    orderBy: Pair<Expression<*>, SortOrder>
+): GroupConcat<T> = GroupConcat(this, separator, distinct, orderBy)
 
-fun<T:Comparable<T>, S:T?> ExpressionWithColumnType<in S>.avg(scale: Int = 2)  : ExpressionWithColumnType<BigDecimal?> = Avg<T, S>(this, scale)
+fun <T : String?> Expression<T>.groupConcat(
+    separator: String? = null,
+    distinct: Boolean = false,
+    orderBy: Array<Pair<Expression<*>, SortOrder>> = emptyArray()
+): GroupConcat<T> = GroupConcat(this, separator, distinct, *orderBy)
 
-fun<T:Any?> ExpressionWithColumnType<T>.stdDevPop(scale: Int = 2) = StdDevPop(this, scale)
+/** Extract a substring from this string expression that begins at the specified [start] and with the specified [length]. */
+fun <T : String?> Expression<T>.substring(start: Int, length: Int): Substring<T> = Substring(this, intLiteral(start), intLiteral(length))
 
-fun<T:Any?> ExpressionWithColumnType<T>.stdDevSamp(scale: Int = 2) = StdDevSamp(this, scale)
+/** Removes the longest string containing only spaces from both ends of string expression. */
+fun <T : String?> Expression<T>.trim(): Trim<T> = Trim(this)
 
-fun<T:Any?> ExpressionWithColumnType<T>.varPop(scale: Int = 2) = VarPop(this, scale)
 
-fun<T:Any?> ExpressionWithColumnType<T>.varSamp(scale: Int = 2) = VarSamp(this, scale)
+// General-Purpose Aggregate Functions
 
-fun<T:Any?> ExpressionWithColumnType<T>.sum() = Sum(this, this.columnType)
+/** Returns the minimum value of this expression across all non-null input values. */
```

Comment:
Maybe it's worth to state what min/max/avg could return null when all of the values of expression are nulls? There were some questions in the past regard to this case.

Reply:
Yes, it may not be obvious at first glance so I'll add that information